eureka2-ext/eureka2-overrides-aws/src/main/java/com/netflix/eureka2/server/service/overrides/S3OutOfServiceOverridesSource.java

```diff
@@ -0,0 +1,141 @@
+package com.netflix.eureka2.server.service.overrides;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.netflix.eureka2.registry.instance.Delta;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.registry.instance.InstanceInfoField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An optimized s3 backed overrides source for OUT_OF_SERVICE status overrides only.
+ * In most cases overrides are only needed for this specific case, and given this constraint,
+ * we can actually make use of s3 object names for storage.
+ *
+ * The file format will be of the form: {config.getPrefix()}/overrideId
```

Comment:
override id == instance id. For clarity what about renaming Overrides.getId() to Overrides.getInstanceId()?